SCons/Tool/MSCommon/common.py

```diff
@@ -31,34 +31,512 @@
 import re
 import subprocess
 import sys
+import glob
 
 import SCons.Util
 
-# SCONS_MSCOMMON_DEBUG is internal-use so undocumented:
-# set to '-' to print to console, else set to filename to log to
-LOGFILE = os.environ.get('SCONS_MSCOMMON_DEBUG')
-if LOGFILE == '-':
+
+class _MSCOMMON:
+
+    # SCONS_MSCOMMON_DEBUG is internal-use so undocumented:
+    # set to '-' to print to console, else set to filename to log to
+    DEBUG_LOGFILE = os.environ.get('SCONS_MSCOMMON_DEBUG')
+
+    # SCONS_MSCOMMON_TRACE is internal-use so undocumented:
+    # set to '-' to print to stderr, else set to filename to log to
+    TRACE_LOGFILE = os.environ.get('SCONS_MSCOMMON_TRACE')
+
+    # bitflags for selectively enabling trace output
+    TRACE_DISPLAY_FUNCTION_ARGLIST   = 0b0001
+    TRACE_DISPLAY_FUNCTION_LOCATION  = 0b0010
+    TRACE_DISPLAY_RETURN_VALUES      = 0b0100
+    TRACE_DISPLAY_ALLFRAMES_ONEBELOW = 0b1000
+
+    # default trace output
+    TRACE_DISPLAY_DEFAULT            = 0b0111
+
+    # Typical usage:
+    #    '0b1111' or '15' display arg lists, file locations, return values, and all frames one below
+    #    '0b0111' or '7'  display arg lists, file locations, and return values [default]
+    #    '0b0101' or '5'  display arg lists, and return values
+    #    '0b0011' or '3'  display arg lists, and file locations
+
+    # SCONS_MSCOMMON_TRACEFLAGS is internal-use so undocumented
+    TRACEFLAGS = int(os.environ.get('SCONS_MSCOMMON_TRACEFLAGS', str(TRACE_DISPLAY_DEFAULT)), 0)
```

Comment:
Change this to be symbolic rather than a bitflag. Having to set 0b0111 is not user friendly.

Reply:
I believe that I have made the change that you have requested.  If not, let me know.

FYI, the response below was not from me.

Reply:
@bdbaddog Now I'm not sure that I have done what you requested. 

 I changed the default trace value to be the bitwise or of the relevant symbolic constants.  However, the environment variable (set from the command line or a batch file) would still need to be a numeric constant (e.g., base 10, base 16, or binary).  

Did you want the _environment variable value_ read to use symbolic constants?  That would be more involved than simply converting the environment variable string value to an integer.

Reply:
See comment below for changes that (hopefully) satisfy your request.